### ISSUE TITLE
Receipt methods return nil

### ISSUE THREAD

**Issue by gerard-morera** [jnbt/candy_check#49]
I am not sure whether I am misusing the gem or there is a Bug.

From `apple_store#verify` I am receiving  an instance of `CandyCheck::AppStoreReceipt`. When I try to access [any of the public methods](https://github.com/jnbt/candy_check/blob/master/lib/candy_check/app_store/receipt.rb) the return is nil.

The reason is that the [read method](https://github.com/jnbt/candy_check/blob/aa190df4ea43ecae61521f129cb4825ad54d2397/lib/candy_check/utils/attribute_reader.rb#L9) does not expect the receipt to be nested:

How the receipt is:
``` 
{"receipt_type"=>"ProductionSandbox",
 "adam_id"=>0,
 "app_item_id"=>0,
 "bundle_id"=>"whatever",

 "in_app"=>
  [{"quantity"=>"1",
    "product_id"=>"some_product",
    "transaction_id"=>"10000006603456836",
    "original_transaction_id"=>"10000006603456836",
}]}
```
So basically to get the `product_id`. Instead of
```
 def read(field)
   attributes['product_id']
end
```
we would need:
```
def  in_app_read(field)
  attributes['in_app'].first['product_id']
end
```
Am I missing anything?

**USAGE VERSION => 0.2.1**

Thanks.

**Comment by jnbt** [comment]
Hi @gerard-morera,

maybe an issue could be that you're not using the deprecated [`transactionReceipt`](https://developer.apple.com/library/ios/documentation/StoreKit/Reference/SKPaymentTransaction_Class/index.html#//apple_ref/occ/instp/SKPaymentTransaction/transactionReceipt)? 

I think this gem only works this these kind of receipts.

**Comment by jnbt** [comment]
I just checked, and for a **old-school** (deprecated) receipts the AppStore API returns data like this: 

```
{
    "receipt": {
        "original_purchase_date_pst": "2020-05-10 11:12:36 America/Los_Angeles",
        "unique_identifier": "UNIQUE_IDENTIFIER",
        "original_transaction_id": "1234567",
        "bvrs": "123456",
        "app_item_id": "1234567",
        "transaction_id": "1234567",
        "quantity": "1",
        "unique_vendor_identifier": "RANDOM-STRING",
        "product_id": "the_product_id",
        "item_id": "1234567",
        "version_external_identifier": "1234567",
        "bid": "the.bundle.id",
        "is_in_intro_offer_period": "false",
        "purchase_date_ms": "1589134356211",
        "purchase_date": "2020-05-10 18:12:36 Etc/GMT",
        "is_trial_period": "false",
        "purchase_date_pst": "2020-05-10 11:12:36 America/Los_Angeles",
        "original_purchase_date": "2020-05-10 18:12:36 Etc/GMT",
        "original_purchase_date_ms": "1589134356211"
    },
    "status": 0
}
```

@gerard-morera I think you're using the more modern (or let us be honest: not fully deprecated) receipt of the app, which include all In-App-Purchases at once. Apple definitely encourages to use this approach. Sadly this gem has still no support for it.

**Comment by gerard-morera** [comment]
Hi, @jnbt thanks for your fast answer. How can I ensure to receive the receipts on the supported format?

**Comment by jnbt** [comment]
@gerard-morera The way I know is the following:

1. Implement the observer pattern for [`paymentQueue:updatedTransactions`](https://developer.apple.com/documentation/storekit/skpaymenttransactionobserver/1506107-paymentqueue?language=objc)
2. Invoke a In-App-Purchase
3. The `paymentQueue:updatedTransactions` will be called with at least one transaction. For every transaction:
    1. Check for it's status to be [`SKPaymentTransactionStatePurchased `](https://developer.apple.com/documentation/storekit/skpaymenttransactionstate/skpaymenttransactionstatepurchased?language=objc)
    2. You get the single transaction receipt via the transactions [`transactionReceipt `](https://developer.apple.com/documentation/storekit/skpaymenttransaction/1617722-transactionreceipt?language=objc)
    3. Encode this data as base64 via [`base64EncodedStringWithOptions`](https://developer.apple.com/documentation/foundation/nsdata/1413546-base64encodedstringwithoptions?language=objc)
    4. Pass this encoded value to your server / this gem
